src/DataGrid.tsx

```diff
@@ -223,10 +225,14 @@ function DataGrid<R, SR, K extends Key>(
   /**
    * defaults
    */
+  const defaultComponents = useDefaultComponents<R, SR>();
   rowHeight ??= 35;
   const headerRowHeight = rawHeaderRowHeight ?? (typeof rowHeight === 'number' ? rowHeight : 35);
   const summaryRowHeight = rawSummaryRowHeight ?? (typeof rowHeight === 'number' ? rowHeight : 35);
-  const RowRenderer = rowRenderer ?? Row;
+  const RowRenderer = components?.rowRenderer ?? defaultComponents?.rowRenderer ?? Row;
+  const headerRenderer = components?.headerRenderer ?? defaultComponents?.headerRenderer;
+  const sortIcon = components?.sortIcon ?? defaultComponents?.sortIcon;
+  const noRowsFallback = components?.noRowsFallback ?? defaultComponents?.noRowsFallback;
```

Comment:
Could use this notation 🤔 
```suggestion
  const noRowsFallback = (components ?? defaultComponents)?.noRowsFallback;
```